src/framework/components/collision/component.js

```diff
@@ -217,21 +258,43 @@ Object.assign(pc, function () {
                 }
             }
 
-            if (this.entity.trigger) {
-                this.entity.trigger.enable();
-            } else if (this.entity.rigidbody) {
+            if (this.entity.rigidbody) {
                 if (this.entity.rigidbody.enabled) {
                     this.entity.rigidbody.enableSimulation();
                 }
+            } else if (this._compoundParent && this !== this._compoundParent) {
```

Comment:
Do we allow `this` to equal `this._compoundParent` ?

Reply:
Actually, seems like it is no more a case, used to be during development. Will test it and remove if needed.

Reply:
Yes, it is possible case. We set any entity with collision component that is related to compound, even entity it self. So when there are changes, we can propagate properly.
It is done to simplify logic, otherwise need separate property for compound collision entity it self, to know when it is changed, etc.